libc/src/stdlib/qsort_r.cpp

```diff
@@ -22,8 +22,14 @@ LLVM_LIBC_FUNCTION(void, qsort_r,
   if (array == nullptr || array_size == 0 || elem_size == 0)
```

Comment:
this file needs to include `heap_sort.h` if it's going to use it.

Reply:
Moved the selection logic into `qsort_util.h`.